module/reports/helper.js

```diff
@@ -0,0 +1,479 @@
+/**
+ * name : helper.js
+ * author : Rakesh
+ * created-date : 01-Nov-2020
+ * Description : Entity report functionalities
+ */
+
+// Dependencies
+
+const UserProjectsHelper = require(MODULES_BASE_PATH + "/userProjects/helper");
+const mongoose = require('mongoose');
+const moment = require('moment');
+
+
+
+/**
+    * Report Helper
+    * @class
+*/
+
+module.exports = class ReportsHelper {
+
+    /**
+    * Entity Report.
+    * @method
+    * @name entity 
+    * @param {String} entityId - mapped entity id.
+    * @param {String} userId - Logged in user id.
+    * @param {String} reportType - report type monthly or quterly.
+    * @param {String} programId - program id
+    * @returns {Object} Entity report.
+   */
+    static entity(entityId = "", userId, reportType, programId = "",getPdf=false) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let query = {};
+                if (entityId) {
+                    query["entityInformation._id"] = mongoose.Types.ObjectId(entityId);
+                } else {
+                    query["userId"] = userId
+                }
+
+                var endOf = "";
+                var startFrom = "";
+                
+                if(reportType == 0) {
+                     let today = moment();
+                     startFrom = today.startOf('week').format('YYYY-MM-DD');
+                     endOf = today.endOf('week').format('YYYY-MM-DD');
+                } else if (reportType == 1) {
+                    endOf = moment().subtract(0, 'months').endOf('month').format('YYYY-MM-DD');
+                    startFrom = moment().subtract(0, 'months').startOf('month').format('YYYY-MM-DD');
+                } else if (reportType == 3) {
+                     startFrom = moment().quarter(moment().quarter()).startOf('quarter').format('YYYY-MM-DD');
+                     endOf = moment().quarter(moment().quarter()).endOf('quarter').format('YYYY-MM-DD');
+                }
+
+                query['$or'] = [
+                    { "lastSync" : { $gte: new Date(startFrom), $lte: new Date(endOf) } },
+                    { "tasks" : { $elemMatch: { lastSync: { $gte: new Date(startFrom), $lte: new Date(endOf) } } } },
+                ]
+
+                if (programId) {
+                    query['programInformation._id'] = mongoose.Types.ObjectId(programId);
```

Comment:
@aman-tunerlabs - Can we please have programId and programExternalId in the outside ?

cc - @rakeshSgr